core/src/main/java/org/elasticsearch/gateway/DanglingIndicesState.java

```diff
@@ -53,7 +57,8 @@
     private final MetaStateService metaStateService;
     private final LocalAllocateDangledIndices allocateDangledIndices;
 
-    private final Map<Index, IndexMetaData> danglingIndices = ConcurrentCollections.newConcurrentMap();
+    private final Map<Index, IndexMetaData> danglingIndices = new HashMap<>();
+    private final Map<Index, IndexMetaData> deletedIndices = new HashMap<>();
```

Comment:
You are correct, one option is that I can change `void findNewAndAddDanglingIndices` to `Map<Index, IndexMetaData> findOnDiskIndices`.... the method would then only add to the `danglingIndices` member variable, and return the `deletedIndices` to be used in the `cleanDeletedIndices` method directly.  What do you think?  I kind of just did it this way to be consistent with the `danglingIndices`... its not the cleanest implementation either way... I mainly did it to avoid having to re-read the on disk indices twice (once to get the dangling, once to get the deleted).